src/PDFs/Tddp4Pdf.cu

```diff
@@ -801,23 +827,12 @@ TDDP4::GenerateSig(unsigned int numEvents) {
     unsigned int tmpparam = parameters;
     wmax = maxWeight;
 
+    exp_functor exp_functor_instance {tmpparam, tmpoff, gammamin, wmax};
```

Comment:
this is not needed, right?    
You are using an unnamed instance in l.835 anyway.